operator/controllers/http/scaler.go

```diff
@@ -0,0 +1,113 @@
+package http
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/kedacore/http-add-on/pkg/k8s"
+	"github.com/kedacore/http-add-on/pkg/util"
+)
+
+const (
+	externalScaler string = "external-scaler"
+	grpc           string = "grpc"
+)
+
+func createOrUpdateExternalScalerResources(
+	ctx context.Context,
+	logger logr.Logger,
+	cl client.Client,
+	httpss metav1.Object,
```

Comment:
@wozniakjan has also an idea on this.

Reply:
WDYM? We do almost the same in KEDA with ScaledObjects and ScaledJobs. Would be enough if I validate the type and return an error if the type isn't correct?

Reply:
I mentioned to @zroubalik earlier that maybe we could explore type aliasing or trivial type definitions. In this case, both `ScalingSets` as well as `ClusterScalingSets` seem to have exactly matching structure so maybe we could do something along the lines of either
```go
type ClusterScalingSet ScalingSet
```
or 
```go
type ClusterScalingSet = ScalingSet
```

I had used the first one with controller-runtime v1 some eons ago successfully and haven't tried it ever since, but could be a good fit for this usecase unless controller-runtime became more strict regarding types.

The implementation went something like this: https://go.dev/play/p/sxZiO_BDw4z

Reply:
I didn't know that this works with kubebuilder but I can give a try because yes. both types are exactly the same, just cluster scoped or namespaced